### ISSUE TITLE
MATLAB AEPsych client: cannot communicate w/ server error

### ISSUE THREAD

**Issue by renata-sadibolova** [facebookresearch/aepsych#253]
We added AEPsych folder and subfolders to Matlab path and get the error below when running the "client=AEPsychClient()"

"Error using AEPsychClient. Cannot create a communication link with the remote server. Please check the input arguments(ADDRESS and PORT) and make sure the server is running. Additional Information: No connection could be made because the target machine actively refused it. See related documentation for troubleshooting steps."

In "related documentation for troubleshooting steps": !ping 192.168.1.111 works fine and so does the "resolvehost". 
Firewall is turned off. This is on Windows 10 machine, Sophos antivirus.
The issue is with connecting to the server - in line 38 in AEPsychClient.m (tcpclient command).
The issue is Matlab-specific (we are able to use AEPsych on Anaconda). We tried 2 Matlab versions: R2021a, R2022b. 

Thanks for your help.

**Comment by mshvartsman** [comment]
* Is the client on the same machine as the server? If they are on different machines you will need to pass the address and port of the server to the client, so replace `client=AEPsychClient()` with `client=AEPsychClient('server_addr', '192.168.1.111')` (and similarly with `port` if you're running it on a port other than 5555. 
* Are you getting any error messages from the server? Can you post server logs?

**Comment by renata-sadibolova** [comment]
Thank you for your prompt response, and the server ip address. Would you be able to give more guidance on the port number please? 

Matlab's help pg. suggests it could be any number from 1 to 65535. I've tried several so far, but none of them worked. Running their example command below connects me to Mathworks server ok, but port 80 does not work for AEPsych server. I'm assuming there's a specific port other than 5555 (line 18 in AEPsychClient script) that goes with the server name '192.168.1.111'.

Matlab's help:
Remote host port, specified as a number between 1 and 65535, inclusive. This property can be set only at object creation.

Example: t = tcpclient("[www.mathworks.com](http://www.mathworks.com/)",80) creates a TCP/IP client connection to port 4012 at [www.mathworks.com](http://www.mathworks.com/).

**Comment by renata-sadibolova** [comment]
We're trying to follow the instructions at https://github.com/facebookresearch/aepsych/tree/main/clients/matlab. 

We added the unzipped AEPsych folder to Matlab's path and tried to run the following command directly in Matlab command window:
client = AEPsychClient('port', 5555);
![Picture1](https://user-images.githubusercontent.com/46522930/226466694-92d8fc74-92a6-4c31-9892-3efb718388ff.png)

**Comment by renata-sadibolova** [comment]
Just to clarify - as per my first post - we did install the server and were able to connect to it on Anaconda (Jupyter) on the same machine (snapshot 1). We tried following instructions on both links in your last response. What's new to add is that there's also an issue w/ connecting to the AEPsych server from Anaconda terminal (snapshot 2 - pls see how it says server is up, waiting for connection... I waited for 15 min and nothing happened). 
Hope that helps clarifying the issue a bit more. Thanks again!

![Picture1](https://user-images.githubusercontent.com/46522930/226749628-1ba28fb6-3982-43a1-afca-b7f226f22e89.png)
![Picture3](https://user-images.githubusercontent.com/46522930/226752135-e4cd9faf-cb02-4fa4-a476-7cf04829cf5b.png)

**Comment by crasanders** [comment]
In your first screenshot, you are running the client and the server in the same Python process. You are creating the server with `server = AEPsychServer(database_path="test.db")` and then passing the server into a client with `client = AEPsychClient(server=server)`. 

In your second screenshot, you are running the server as a standalone program. The logs indicate that the server has started and is waiting for a client to connect. If you run `client=AEPsychClient()` in MATLAB while the server is running, you should see some log messages indicating that the client has connected.

**Comment by renata-sadibolova** [comment]
Thanks. Still no luck with: client=AEPsychClient() in MATLAB while the server is running. For clarity - I created the server with server = AEPsychServer(database_path="test.db") in Jupyter (as per my snapshot above). First, I tried client=AEPsychClient() in MATLAB immediately after. I then passed the server into a client again in Jupyter, with  client = AEPsychClient(server=server) - same snapshot and same result = no complaints/error messages. I then tried the MATLAB command again. The MATLAB error message was the same both times (as on my very first snapshot in this thread)...
